### DISCUSSION TITLE
computeRequiredInitialRootTileKeys

### DISCUSSION THREAD

**Discussion by blueRaining** [heremaps/harp.gl#2233] [Q&A] [answered]
hello,
   i like harpgl very much,and i want to read the source code about it,how  can i understand the function of  computeRequiredInitialRootTileKeys,when i set the offset is always zeor,it always Operates normally. If I can get the source-related documents, it will be my honor. i like harp very much

**Comment by nzjony** [comment, marked as answer]
@blueRaining , sorry for the delay.

Yes, this function may be confusing, sorry about that.

Basically, the function just estimates what initial root tile keys are in the viewport, it takes a very approximate approach, so it overestimates, but it is not expensive, so we didn't have a more complicated solution.

It is basically used to show the multiple worlds in a flat planar projection when you zoom out. I.e. for a globe projection, returning the tile key with 0 offset is enough. The offset just means which world used, see screenshot.

Regarding documentation, we don't have anything for this concept unfortunately.

<img width="1680" alt="Screenshot 2021-07-19 at 11 44 56" src="https://user-images.githubusercontent.com/2533428/126140744-8f117176-0130-4492-8111-631071e3e925.png">

Does that answer your questions, or do you need any more help?
